examples/applications/plot_time_series_lagged_features.py

```diff
@@ -0,0 +1,433 @@
+# predicted from the features. It can be influenced by other variables
+# not properly captured by the existing lagged features. For instance
+# whether or not it will rain in the next hour cannot be fully be anticipated
+# from the past hours bike rental data. This is what we call aleatoric
+# uncertainty.
+#
+# Quantile regression makes it possible to give a finer description of that
+# distribution without making strong assumptions on its shape.
+#
+# The conditional 5th percentile (a.k.a. 0.05-quantile) can be estimated with:
+gbrt_percentile_05 = HistGradientBoostingRegressor(loss="quantile", quantile=0.05)
+evaluate(gbrt_percentile_05, X, y, cv=ts_cv)
+
+# %%
+# The conditional median (0.50-quantile) can be estimated with:
+gbrt_median = HistGradientBoostingRegressor(loss="quantile", quantile=0.5)
+evaluate(gbrt_median, X, y, cv=ts_cv)
+
+# %%
+# And finally the 0.95 quantile:
+gbrt_percentile_95 = HistGradientBoostingRegressor(loss="quantile", quantile=0.95)
+evaluate(gbrt_percentile_95, X, y, cv=ts_cv)
+
+# %%
+# A Qualitative Look at the Predictions
+# -------------------------------------
+# We can now visualize the performance of the model with regards
+# to the 5th percentile, median and the 95th percentile:
+all_splits = list(ts_cv.split(X, y))
+train_idx, test_idx = all_splits[0]
+
+X_train, X_test = X.iloc[train_idx], X.iloc[test_idx]
+y_train, y_test = y.iloc[train_idx], y.iloc[test_idx]
+
+max_iter = 50
+gbrt_mean_poisson = HistGradientBoostingRegressor(loss="poisson", max_iter=max_iter)
+gbrt_mean_poisson.fit(X_train, y_train)
+mean_predictions = gbrt_mean_poisson.predict(X_test)
+
+gbrt_median = HistGradientBoostingRegressor(
+    loss="quantile", quantile=0.5, max_iter=max_iter
+)
+gbrt_median.fit(X_train, y_train)
+median_predictions = gbrt_median.predict(X_test)
+
+gbrt_percentile_5 = HistGradientBoostingRegressor(
+    loss="quantile", quantile=0.05, max_iter=max_iter
+)
+gbrt_percentile_5.fit(X_train, y_train)
+percentile_5_predictions = gbrt_percentile_5.predict(X_test)
+
+gbrt_percentile_95 = HistGradientBoostingRegressor(
+    loss="quantile", quantile=0.95, max_iter=max_iter
+)
+gbrt_percentile_95.fit(X_train, y_train)
+percentile_95_predictions = gbrt_percentile_95.predict(X_test)
+
+last_hours = slice(-96, None)
+fig, ax = plt.subplots(figsize=(15, 7))
+plt.title("Predictions by regression models")
+ax.plot(
+    y_test.values[last_hours],
+    "x-",
+    alpha=0.2,
+    label="Actual demand",
+    color="black",
+)
+ax.plot(
+    median_predictions[last_hours],
+    "^-",
+    label="GBRT median",
+)
+ax.plot(
+    mean_predictions[last_hours],
+    "x-",
+    label="GBRT mean (Poisson)",
+)
+ax.fill_between(
+    np.arange(96),
+    percentile_5_predictions[last_hours],
+    percentile_95_predictions[last_hours],
+    alpha=0.3,
+    label="GBRT 90% interval",
+)
+_ = ax.legend()
+
+# %%
+# Looking at the performance of non-linear regression models vs
+# perfect models:

```

Comment:
There are no "perfect models" directly displayed on those plots. For instance, the "perfect" models for the 5%-percentile estimation would have 95% of the points above the diagonal and 5% below. Furthermore, those fractions should stay uniform when binning horizontally. But it's hard to see if our models are good or not in that respect using those plots. Maybe they are maybe not.